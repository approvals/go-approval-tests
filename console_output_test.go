package approvals

import (
	"fmt"
	"os"
	"testing"
	"time"
)

func TestVerifyOutput(t *testing.T) {
	console := NewConsoleOutput()

	fmt.Print("Hello, World!")
	
	console.Close()
	
	output := console.GetOutput()
	t.Logf("Captured output: '%s'", output)
	
	console.VerifyOutput(t)
}

func TestVerifyError(t *testing.T) {
	console := NewConsoleOutput()
	defer console.Close()

	fmt.Fprint(os.Stderr, "Error message!")
	
	time.Sleep(10 * time.Millisecond)
	
	console.VerifyError(t)
}

func TestVerifyAll(t *testing.T) {
	console := NewConsoleOutput()
	defer console.Close()

	fmt.Print("Standard output")
	fmt.Fprint(os.Stderr, "Error output")
	
	time.Sleep(10 * time.Millisecond)
	
	console.VerifyAll(t)
}
