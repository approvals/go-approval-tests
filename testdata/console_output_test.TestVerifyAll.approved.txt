STDOUT:
Standard output
STDERR:
Error output
